sphinx4-core/src/main/java/edu/cmu/sphinx/api/FeatureExtractor.java

```diff
@@ -0,0 +1,146 @@
+package edu.cmu.sphinx.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import edu.cmu.sphinx.frontend.Data;
+import edu.cmu.sphinx.frontend.DataEndSignal;
+import edu.cmu.sphinx.frontend.DataProcessingException;
+import edu.cmu.sphinx.frontend.DoubleData;
+import edu.cmu.sphinx.frontend.FloatData;
+import edu.cmu.sphinx.frontend.FrontEnd;
+import edu.cmu.sphinx.frontend.util.StreamDataSource;
+import edu.cmu.sphinx.util.props.ConfigurationManager;
+
+/**
+ * Extracts features from input stream
+ * 
+ * @author Vladisav Jelisavcic
+ * 
+ */
+class FeatureExtractor {
+    private FrontEnd frontEnd;
+    private StreamDataSource audioSource;
+    private int featureLength = -1;
+
+    /** The logger for this class */
+    private static final Logger logger = Logger
```

Comment:
Sphinx4 uses it's own logging framework, it is better to stick to that one. In high-level API we do not use logging.